packages/compose/src/utils/create-higher-order-component/index.ts

```diff
@@ -30,12 +19,11 @@ export type HigherOrderComponent< HOCProps extends Record< string, any > > = <
  *
  * @return Component class with generated display name assigned.
  */
-function createHigherOrderComponent<
-	HOCProps extends Record< string, any > = {}
->( mapComponent: HigherOrderComponent< HOCProps >, modifierName: string ) {
-	return < InnerProps extends HOCProps >(
-		Inner: ComponentType< InnerProps >
-	) => {
+export function createHigherOrderComponent(
+	mapComponent: ( Inner: ComponentType< any > ) => ComponentType< any >,
+	modifierName: string
+): typeof mapComponent {
```

Comment:
Now when `createHigherOrderComponent` and `mapComponent` are no longer generic functions, with type parameters, that means that `createHigherOrderComponent` erases the type of `mapComponent` and returns a generic `any => any` mapper.

For example, if I pass `ComponentType<MyProps>` to `pure`, I should get a `ComponentType<MyProps>`, component of the same type. But I get `ComponentType<any>`.